src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -102,6 +118,32 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     userManager.updateUser(updatedUser);
+    String tags = (String) request.getParameter("tags");
+    if (tags != null) {
+      updateTagsFollowedBy(id, tags);
+    }
+
     response.sendRedirect("/user/" + user.id);
   }
+
+  private void updateTagsFollowedBy(long userId, String tagsString) {
+    String[] tagNames = getTagNames(tagsString);
+    List<Long> tagIds = new ArrayList<>();
+    for (String tagName : tagNames) {
+      tagIds.add(tagManager.readOrCreateTagByName(tagName).id);
+    }
+
+    followManager.updateFollowedTagIds(userId, tagIds);
+  }
+
+  /**
+   * Returns an array of tag names parsed from the tagsArrayString.
+   *
+   * @param tagsArrayString a string representation of a tag names array.
+   * @return an array of tag names.
+   */
+  private String[] getTagNames(String tagsArrayString) {
+    String[] tagNames = tagsArrayString.split(",");
+    return tagNames;
+  }
```

Comment:
This method can just be inlined, I don't see it being used multiple times and it's not overly complicated.